bug fix: remove chunks when interruption is requested inside concatenation

bruno-at-orange opened the pull request:
A simple fix: temporary files are cleaned when there is an error. User interruption wasn't an error, it is  now.
closing #466 

2 extra things:
- Remove `bRemoveChunks` from the concatenate API , because it is always used with `bRemoveChunks=true`
- Reserve the size of the output file in concatenation.

Fix bug #468 
:warning: Don't forget to add a line in TestKhiops/NewV10/DetectFileFormatSideEffects/results.ref/D_NoFieldsHL.txt

bruno-at-orange committed:
bug fix in concatenation: remove chunks when interruption is requested

marcboulle reviewed (changes requested):
LGTM

Pour les deux questions, cela me semble pertinent de 
- supprimer le paramètre inutile bRemoveChunks
- implémenter la réserve de la taille

bruno-at-orange committed:
Simplify concatenate API

Remove never used parameter bRemoveChunks in the concatenate method

bruno-at-orange committed:
Reserve size for output file in concatenation

bruno-at-orange committed:
Add Set/GetHeaderLineUsed in concatenate API

It allows the distinction between no header line and the empty header line (an empty line)

marcboulle reviewed (approved):
LGTM
